### PR TITLE
feat: add option to not transform processes on win.SetVisibleOnAllWorkspaces

### PR DESCRIPTION
**Pull request by CyrusRoshan**
#### Description of Change

Allows calls to win.SetVisibleOnAllWorkspaces to skip transforming the process type. This allows us to use the same behavior found in Electron versions 9.2.1 and under, which is useful for "pinning" an app to a workspace/desktop, without causing other issues related to process transformation. 

This PR adds optional backwards-compatibility with 9.2.1 and under, while maintaining the same default behavior.

Fixes: (#27101)
Process transformation introduced in this PR: [PR #24956](https://github.com/electron/electron/pull/24956)

Stakeholders:
@codebytere 
@zcbenz 
@MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Allowed skipping process type transformation in win.SetVisibleOnAllWorkspaces on macOS.

### CODE REVIEW

**Review by zcbenz — APPROVED** [review]

**Comment by zcbenz** [comment]
This PR requires approval from @electron/wg-api since it adds a new option.

**Comment by CyrusRoshan** [comment]
Sweet, thank you! Let me know if there's anything I can do to help.

**Review by codebytere — APPROVED** [review]

File: docs/api/browser-window.md [hunk 1]
```diff
@@ -1626,6 +1626,8 @@ Returns `Boolean` - Whether the menu bar is visible.
 * `options` Object (optional)
   * `visibleOnFullScreen` Boolean (optional) _macOS_ - Sets whether
     the window should be visible above fullscreen windows
+  * `skipTransformProcessType` Boolean (optional) _macOS_ - Sets whether
+    the process type should be transformed. Default is `false`
```
Comment (jkleinsc) [inline docs/api/browser-window.md:1630]: While functionally true, this description doesn't seem really helpful to developers.  Can you reword to provide better insight to developers as to why you would want to set this?

**Review by jkleinsc — CHANGES_REQUESTED** [review]
Adding this option looks good to me, but the documentation needs to be clearer as to the functionality it provides

File: docs/api/browser-window.md (hunk 1 above)
Comment (CyrusRoshan) [reply docs/api/browser-window.md]: Sure! How does the updated description sound?

File: docs/api/browser-window.md [hunk 2]
```diff
@@ -1625,7 +1625,10 @@ Returns `Boolean` - Whether the menu bar is visible.
 * `visible` Boolean
 * `options` Object (optional)
   * `visibleOnFullScreen` Boolean (optional) _macOS_ - Sets whether
-    the window should be visible above fullscreen windows
+    the window should be visible above fullscreen windows.
+  * `skipTransformProcessType` Boolean (optional) _macOS_ - Transforming
+    the process type must hide the window and dock for a short time. This
+    can be skipped if the process type has already been transformed.
```
Comment (jkleinsc) [inline docs/api/browser-window.md:1631]:
```suggestion
  * `skipTransformProcessType` Boolean (optional) _macOS_ - By default on macOS, setVisibleOnAllWorkspaces transforms the process type in order to hide the window and dock for a short time.  If this is undesirable, setting this value to true will override that behavior.
```

**Review by jkleinsc — CHANGES_REQUESTED** [review]
@CyrusRoshan the updated description is a little more helpful but it is still unclear.  I'm still not 100% clear why we are changing the process type and I think that's the part that needs clarifying.  Maybe including more context from #27101 would be helpful.  I'm trying to think of developers who don't have the context that you do if that makes sense.

**Review by jkleinsc — APPROVED** [review]
@CyrusRoshan thanks for updating the description.  It is much clearer now what this option does.

**Comment by zcbenz** [comment]
This PR has got enough approvals from members of wg-api, it is a bug of bot that `api-review/approved ✅ ` label can not be applied.

**Pull request merged**